lib/tasks/repair_nfn.rake

```diff
@@ -0,0 +1,43 @@
+namespace :data_repair do
+  task :nfn do
+    query = %q(
+      select user_id, reducer_key, subgroup
+      from user_reductions
+      where reducible_type='Project' and reducible_id=1558
+      group by reducible_type, reducible_id, reducer_key, user_id, subgroup
+      having count(*) > 1
+    )
+
+    affected_items = ActiveRecord::Base::Connection.execute(query)
+
+    affected_items.each do |params|
+      user_id,  reducer_key, subgroup = params
+
+      ActiveRecord::Base.transaction do
+        reductions = UserReductions.where(
+          reducible_type: 'Project',
+          reducible_id: 1558,
+          user_id: user_id,
+          reducer_key: reducer_key,
+          subgroup: subgroup
+        ).lock(true)
```

Comment:
I can't find any documentation for this method. https://api.rubyonrails.org/classes/ActiveRecord/Transactions/ClassMethods.html doesn't have it, and the model/object instance pessimistic locking uses either `.with_lock` or `.lock!`, as here: https://api.rubyonrails.org/classes/ActiveRecord/Locking/Pessimistic.html

Reply:
Unless I'm confused, this is just a more verbose version of

```ruby
shugo = Account.where("name = 'shugo'").lock(true).first
```

from the docs (albeit not restricting ourselves to only locking one row).

Reply:
My bad, I was reading `.lock` as being called on the transaction and thrown off by `(true)` instead of using the bang method. That should work.

Reply:
I think that the exact method being called is `ActiveRecord::QueryMethods::WhereChain#lock`

Reply:
As soon as I read your comment I looked at the source and did the same thing, lol